node/test/util/commit.js

```diff
@@ -3385,3 +3385,74 @@ x=U:Cmsg\n#x-2 s=Sa:s;Bmaster=x;Os Cmsg\n#s-1 README.md=foo,addedbyhook=bar`;
 
     });
 });
+
+describe("commit mid-merge", function() {
+    it("handles a commit mid-merge", co.wrap(function*() {
+        const MergeUtil    = require("../../lib/util/merge_util");
+        const MergeCommon  = require("../../lib/util/merge_common");
+        const Open         = require("../../lib/util/open");
+        const SeqStateUtil = require("../../lib/util/sequencer_state_util");
+
+        const start = `a=B:C2-1 a=a1;
+                           C3-2 a=a3;
+                           C4-2 a=a4;
+                           Bb3=3;Bb4=4;|
+              x=S:Cm2-1 s=Sa:2;
+                  Cm3-m2 s=Sa:3;
+                  Cm4-m2 s=Sa:4;
+                  Bmaster=m3;Bm4=m4;`;
+        const repoMap = yield RepoASTTestUtil.createMultiRepos(start);
+        const repo = repoMap.repos.x;
+
+        const m4 = yield NodeGit.Commit.lookup(repo,
+                                               repoMap.reverseCommitMap.m4);
+        try {
+            yield MergeUtil.merge(repo,
+                                  null,
+                                  m4,
+                                  MergeCommon.MODE.NORMAL,
+                                  Open.SUB_OPEN_OPTION.FORCE_OPEN,
+                                  [],
+                                  "msg",
+                                  function() {});
+        } catch (e) {
+            // we expect the merge to fail, but that's OK
```

Comment:
why is merge fail expected? Any chance we can create a test case that merge can finish successfully?

Reply:
This is testing the case that you are mid-merge (that is, the merge has failed and you are now doing conflict resolution).